File: array/manipulation.py
#!/usr/bin/env python

n, m = map(int, input().split())
arr = []

for i in range(n + 10):
	arr.append(0)

max = 0
for i in range(m):
	a,b,k = map(int, input().split())
	for j in range(a, b+1):
		arr[j] += k
		if arr[j] >= max:
			max = arr[j]

print(max)

File: array/manipulation.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
	long long int n, m;
	long long int max = 0;
	long long int a, b, k;

	cin >> n >> m;
	long int *arr=new long int[n+1]();

	while(m--)
	{
		cin >> a >> b >> k;
		arr[a] += k;
		if(b + 1 <= n)
			arr[b + 1] -= k;
	}

	long int x = 0;
	for (long int i = 1; i <= n; ++i)
	{
		x += arr[i];
		if(x >= max)
			max = x;
	}

	cout << max << endl;

	return 0;
}